### ISSUE TITLE
Uncommon.Cs.ct_find_uint8 is not constant-time

### ISSUE THREAD

**Issue by cfcs** [mirage/mirage-crypto#23]
The code currently looks like this (I've inserted comments):
```ocaml
  let ct_find_uint8
    ?(off=0)  (* branch on value of [off: int option] *)
    ~f cs =
    let rec go acc i = function
      | 0 -> acc
      | n ->
          let acc = match (acc, f (get_uint8 cs i))
           with 
(* [match] introduces a branch; since we branch on a sensitive value this
means we leak timing of whether or not [f] is true.
Example: If we always take one branch, the branch predictor will be
  hot for our code.
If we find something in the middle, we disrupt that.
Depending on at what offset we find a match, we may or may not prime
 the branch predictor again. That's worse than just leaking the predicate,
since it can leak the index of the the matching predicate.

             *)
            | (None, true) ->

              Some i (* allocating a new Some box is observable,
                     so we leak the information that predicate [f] is true *)
            | _            -> acc in
          go acc (succ i) (pred n) in
    go None off
    (len cs - off) (* if [off] is > len cs the loop never terminates *)
```

@dinosaure Should we write a constant-time version of this for `eqaf`? (e.g. take non-optional offset, always update `acc` using constant-time `select` (maybe we should also expose our constant-time `select` code)?

EDIT(s): Formatting + elaboration

**Comment by dinosaure** [comment]
`eqaf` was just released into ocaml/opam-repository#15982 with `select_int`, `one_if_not_zero` and `zero_if_not_zero` 🎉

**Comment by hannesm** [comment]
@dinosaure cool, once that is in opam-repository, feel free to PR here changes to use it accordingly.

**Comment by hannesm** [comment]
this has been addressed by #52, thanks.
